lambdas/services/base/iam_service.py

```diff
@@ -0,0 +1,32 @@
+import boto3
+from botocore.exceptions import ClientError
+from utils.audit_logging_setup import LoggingService
+
+logger = LoggingService(__name__)
+
+
+class IAMService:
+    def __init__(self):
```

Comment:
I wonder if we should configure this to be a singleton also? For now it's only the s3 service using it, however because the s3 service is a singleton it will spawn multiple instances of this iam service when we only need one